Tests/FluentAssertions.Specs/Formatting/FormatterSpecs.cs

```diff
@@ -216,6 +216,119 @@ public void When_the_object_is_a_generic_type_without_custom_string_representati
+                .Should().Match($"*but found FluentAssertions.Specs.Formatting.FormatterSpecs+StuffRecord{Environment.NewLine}{{*")
+                .And.Match(
+                    $"*, {Environment.NewLine}" +
+                    $"    SingleChild = FluentAssertions.Specs.Formatting.FormatterSpecs+ChildRecord{Environment.NewLine}" +
+                    $"    {{{Environment.NewLine}" +
+                    $"*");
+    }
+
+    [Fact]
+    public void When_the_object_is_an_anonymous_type_it_should_show_the_properties_recursively()
+    {
+        // Arrange
+        var stuff = new
+        {
+            Description = "absent",
+            SingleChild = new { ChildId = 8 },
+            Children = new[] { 1, 2, 3, 4 },
+        };
+
+        var expectedStuff = new
+        {
+            SingleChild = new { ChildId = 4 },
+            Children = new[] { 10, 20, 30, 40 },
+        };
+
+        // Act
+        Action act = () => stuff.Should().Be(expectedStuff);
+
+        // Assert
+        act.Should().Throw<XunitException>()
+            .And.Message
+                .Should().Match($"*Expected stuff to be {Environment.NewLine}{{*")
+                .And.Match(@"*Description = ""absent""*")
+                .And.Match($"**SingleChild = {Environment.NewLine}    {{*")
+                .And.Match("*ChildId = 4*")
+                .And.Match("*Children = {10, 20, 30, 40}*")
+                .And.NotContain("AnonymousType");
+    }
+
+    [Fact]
+    public void When_the_object_is_an_empty_anonymous_type_it_should_show_braces_on_the_same_line()
+    {
+        // Arrange
+        var stuff = new
+        {
+        };
+
+        // Act
+        Action act = () => stuff.Should().BeNull();
+
+        // Assert
+        act.Should().Throw<XunitException>()
+            .Which.Message.Should().Match("*but found { }*")
+;
+    }
+
+    [Fact]
+    public void When_the_object_is_a_tuple_it_should_show_the_properties_recursively()
+    {
+        // Arrange
+        (int TupleId, string Description, List<int> Children) stuff = (1, "description", new() { 1, 2, 3, 4 });
+
+        (int, string, List<int>) expectedStuff = (2, "WRONG_DESCRIPTION", new List<int> { 4, 5, 6, 7 });
+
+        // Act
+        Action act = () => stuff.Should().Be(expectedStuff);
+
+        // Assert
+        act.Should().Throw<XunitException>()
+            .WithMessage("*1*description*{1, 2, 3, 4}*")
+            .WithMessage("*2*WRONG_DESCRIPTION*{4, 5, 6, 7}*")
+            .And.Message.Should().NotContain("System.ValueTuple");
+    }
+
+    [Fact]
+    public void When_the_object_is_a_record_it_should_show_the_properties_recursively()
+    {
+        // Arrange
+        var stuff = new StuffRecord(
+            RecordId: 9,
+            RecordDescription: "descriptive",
+            SingleChild: new(ChildRecordId: 80),
+            RecordChildren: new() { 4, 5, 6, 7 });
+
+        var expectedStuff = new
+        {
+            RecordDescription = "WRONG_DESCRIPTION",
```

Comment:
```suggestion
            RecordDescription = "WRONG_DESCRIPTION"
```

Reply:
I always go the other way and encourage the extra comment.  My rationale is that it keeps diffs cleaner when extra lines are added - if the comment is already there then line 317 (in this example) won't show up as a difference.  

Also I can simply duplicate lines without having to worry about adding/removing commas.

Reply:
I now see we don't enforce this anywhere, so just scratch this comment.
I agree about the benefits of keeping commas, I guess it's just me that expects something more to come when I see a comma.

> Mozart’s kids would taunt him by playing incomplete scales on the piano forcing him to rush downstairs and complete them.